Client/src/Remote/FluentRequest.tsx

```diff
@@ -0,0 +1,93 @@
+import SnackbarUtils from "../Components/Utils/SnackbarUtils"
+import { stringify } from "query-string"
+
+const requestBase: RequestInit = {
+  mode: 'cors', // no-cors, *cors, same-origin
+  cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
+  credentials: 'same-origin', // include, *same-origin, omit
+  redirect: 'follow', // manual, *follow, error
+  referrerPolicy: 'no-referrer', // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
+}
+
+export const get = (route: string): FluentRequest => {
+  return new FluentRequest('GET', route)
+}
+
+export const post = (route: string): FluentRequest => {
+  return new FluentRequest('POST', route)
+}
+
+export const put = (route: string): FluentRequest => {
+  return new FluentRequest('PUT', route)
+}
+
+//underscore is needed so that JS does not confuse it with reserved delete keyword
+export const _delete = (route: string): FluentRequest => {
+  return new FluentRequest('DELETE', route)
+}
+
+export const patch = (route: string): FluentRequest => {
+  return new FluentRequest('PATCH', route)
+}
+
+export class FluentRequest {
+  private url: string
+  private query: string
+  private requestInit: RequestInit
+  private headers: Headers
+
+  constructor(method, route) {
+    this.requestInit = { ...requestBase }
+    this.requestInit.method = method
+    this.url = route
+    this.headers = new Headers()
+    this.headers.append('Content-Type', 'application/json')
+  }
+
+  public withFile(file: File): FluentRequest {
```

Comment:
Big fan of how you did this

Reply:
thanks for the feedback this is much cleaner compared with the first imlpementation